Sources/Controls/UIBarButtonItem+Combine.swift

```diff
@@ -0,0 +1,69 @@
+//
+//  UIBarButtonItem+Combine.swift
+//  CombineCocoa
+//
+//  Created by Shai Mishali on 12/08/2019.
+//  Copyright © 2019 Shai Mishali. All rights reserved.
+//
+
+import Combine
+import UIKit
+
+public extension UIBarButtonItem {
+  /// A publisher which emits whenever this UIBarButtonItem is tapped.
+  var tapPublisher: AnyPublisher<Void, Never> {
```

Comment:
Thought - should we send the bar button item as the value? Isn't that what RxCocoa does (I could go look but I don't have time... I know that's lame!). It can be a useful thing to have when doing something with the event, and similar to the `sender` parameter on Cocoa event handlers.

Reply:
This is the equivalent in RxCocoa.
https://github.com/ReactiveX/RxSwift/blob/6b2a406b928cc7970874dcaed0ab18e7265e41ef/RxCocoa/iOS/UIBarButtonItem%2BRx.swift#L33

Reply:
I prefer to not relay UI Components over a publisher, it feels wrong to me personally :)

Reply:
PS weak must be a var and optional. 

`'weak' must be a mutable variable, because it may change at runtime`